libavogadro/src/extensions/yaehmop/yaehmopextension.cpp

```diff
@@ -0,0 +1,603 @@
+/******************************************************************************
+
+  This source file is part of the Avogadro project.
+
+  Copyright 2016 Kitware, Inc.
+
+  This source code is released under the New BSD License, (the "License").
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+******************************************************************************/
+
+#include <avogadro/molecule.h>
+#include <avogadro/atom.h>
+
+#include <avogadro/plotwidget.h>
+#include <avogadro/plotobject.h>
+#include <avogadro/plotaxis.h>
+#include <avogadro/plotpoint.h>
+
+#include <openbabel/generic.h>
+#include <openbabel/mol.h>
+
+#include <QAction>
+#include <QDebug>
+#include <QDialogButtonBox>
+#include <QDir>
+#include <QFileDialog>
+#include <QLayout>
+#include <QList>
+#include <QMessageBox>
+#include <QPair>
+#include <QProcess>
+#include <QString>
+#include <QTextEdit>
+#include <QVector3D>
+
+#include "yaehmopbanddialog.h"
+#include "yaehmopextension.h"
+#include "yaehmopout.h"
+
+using namespace std;
+using namespace Eigen;
+
+namespace Avogadro
+{
+
+  YaehmopExtension::YaehmopExtension(QObject *parent) : Extension(parent)
+  {
+    // create an action for our first action
+    QAction *action = new QAction( this );
+    action->setText(tr("Calculate Band Structure"));
+    m_actions.append(action);
+    connect(action, SIGNAL(triggered()), SLOT(calculateBandStructure()));
+
+// These will be added soon!
+/*
+    // create an action for our second action
+    action = new QAction(this);
+    action->setText(tr("Plot Total Density of States"));
+    m_actions.append(action);
+    connect(action, SIGNAL(triggered()), SLOT(plotTotalDOS()));
+
+    // create an action for our third action
+    action = new QAction( this );
+    action->setText( tr("Plot Partial Density of States"));
+    m_actions.append(action);
+    connect(action, SIGNAL(triggered()), SLOT(plotPartialDOS()));
+*/
+    action = new QAction( this );
+    action->setText( tr("Set Parameters File"));
+    m_actions.append(action);
+    connect(action, SIGNAL(triggered()), SLOT(setParametersFile()));
+
+    action = new QAction( this );
+    action->setText(tr("Execute Custom Input"));
+    m_actions.append(action);
+    connect(action, SIGNAL(triggered()), SLOT(executeCustomInput()));
+  }
+
+  YaehmopExtension::~YaehmopExtension()
+  {
+  }
+
+  QList<QAction *> YaehmopExtension::actions() const
+  {
+    return m_actions;
+  }
+
+  QString YaehmopExtension::menuPath(QAction *action) const
+  {
+    return tr("E&xtensions") + '>' + tr("Yaehmop");
```

Comment:
Can we make sure to add a separator before this? Take a look at the force field extension for an example. Otherwise we look like part of the "spectra" section.